framework/src/actions/CreateProblemAction.C

```diff
@@ -68,6 +69,28 @@ CreateProblemAction::act()
   {
     // build the problem only if we have mesh
     {
+      // if users try to use Eigen Executioner and do not specify a problem type,
+      // we should create EigenProblem
+      if (_app.useEigenExecutioner() && _type == "FEProblem")
+      {
+        _type = "EigenProblem";
+        parameters().set<std::string>("type") = "EigenProblem";
+        _moose_object_pars =
+            (!parameters().have_parameter<bool>("skip_param_construction") ||
+                     (parameters().have_parameter<bool>("skip_param_construction") &&
+                      !parameters().get<bool>("skip_param_construction"))
+                 ? _factory.getValidParams(_type)
+                 : validParams<MooseObject>());
```

Comment:
This is such a nasty line of code that it really shouldn't be repeated (copied). At the very least, it should be factored into a helper method in the original class and called here.